OpenRA.Game/Traits/Player/FrozenActorLayer.cs

```diff
@@ -27,10 +27,13 @@ public class FrozenActorLayerInfo : Requires<ShroudInfo>, ITraitInfo
 		public object Create(ActorInitializer init) { return new FrozenActorLayer(init.Self, this); }
 	}
 
-	public class FrozenActor
+	public class FrozenActor : IOccupySpace
```

Comment:
Implementing this (a trait-specific interface) here doesn't feel right to me.

Can you please instead change this to:
* Rename `Footprint` in here to `ProjectedFootprint`.
* Introduce a new `CPos[] Footprint` member that is assigned in the constructor (pass it in from `FrozenUnderFog` where it is already available as an intermediate variable in calculating the `PPos[]` footrpint)
* Write a new `public static CPos NearestCellTo(this FrozenActor frozen, CPos other)` extension method in `IOccupySpaceExts` that enumerates the footprint in the same way it does for `IOccupySpace`
* Change `ExternalCapture` to call the appropriate extension method for actor vs frozen actor.